r/tests/testthat/test-metadata.R

```diff
@@ -231,6 +231,9 @@ test_that("metadata of list elements (ARROW-10386)", {
     "Row-level metadata is not compatible with datasets and will be discarded"
   )
 
+  # Reset directory as previous write will have created some files and the defualt
```

Comment:
So this is configurable and we've changed the default? Can we set the default in the R bindings to be the status quo, just to eliminate surprises?

Also:
```suggestion
  # Reset directory as previous write will have created some files and the default
```

Reply:
Yes and no.  It's configurable (in `FileSystemDatasetWriteOptions` you would set `existing_data_behavior` to `kOverwriteOrIgnore`) but I have not included exposing those options to R as part of this PR.  I'll add this request to ARROW-13703 if that works for you.